service-base/src/main/java/org/eclipse/hono/service/metric/Metrics.java

```diff
@@ -82,6 +83,30 @@ void reportTelemetry(
             int payloadSize,
             Sample timer);
 
+    /**
+     * Reports a telemetry message or event received from a device.
+     *
+     * @param type The type of message received, e.g. <em>telemetry</em> or <em>event</em>.
+     * @param tenantId The tenant that the device belongs to.
+     * @param tenantObject The tenant configuration object.
+     * @param outcome The outcome of processing the message.
+     * @param qos The delivery semantics used for sending the message downstream.
+     * @param payloadSize The number of bytes contained in the message's payload.
+     * @param timer The timer indicating the amount of time used
+     *              for processing the message.
+     * @throws NullPointerException if any of the parameters are {@code null}.
+     * @throws IllegalArgumentException if type is neither telemetry nor event or
+     *                    if payload size is negative.
+     */
+    void reportTelemetry(
+            MetricsTags.EndpointType type,
+            String tenantId,
```

Comment:
IMHO we do not need this if we also include the `TenantObject`, right?

Reply:
There is a scenario where there can be issue retrieving `TenantObject`using `getTenantConfiguration(...)`. In this case `TenantObject` will be null and tenant Id is still required to report the metrics.